### PR TITLE
Added feature to pass kwargs to the build process for docker_image.pr…

### PR DESCRIPTION
**Pull request by pabloh007**
…esent #42606

### What does this PR do?
It adds a feature where you can now pass keyword arguments on the salt state for the build process. It also checks the arguments available for the build function "only" so it other kwargs are passed that are not from the build function, they will be ignored.
### What issues does this PR fix or reference?
Previously you could not pass keyword args to the `docker.build`  
### New Behavior
you can now pass keyword args through docker_image.present state

web_service_img:0.0.5:
  docker_image.present:
    - build: /usr/share/dockerfiles/web_service
    - kwargs:
        buildargs:
           APP_VERSION: 0.0.5

### Tests written?
No

### CODE REVIEW

File: salt/states/docker_image.py [hunk 1]
```diff
@@ -40,6 +40,7 @@
 
 # Import salt libs
 import salt.utils.docker
+import salt.utils
```
Comment (terminalmage) [inline salt/states/docker_image.py]: Please change this to an import of `salt.utils.args`.

**Review by terminalmage — CHANGES_REQUESTED** [review]

**Comment by pabloh007** [comment]
Update is done. Thanks!

File: salt/states/docker_image.py [hunk 2]
```diff
@@ -40,6 +40,7 @@
 
 # Import salt libs
 import salt.utils.docker
+import salt.utils.args
```
Comment (rallytime) [inline salt/states/docker_image.py]: Can you add the following import here:
```
from salt.ext.six.moves import zip
```

File: salt/states/docker_image.py [hunk 3]
```diff
@@ -202,10 +203,19 @@ def present(name,
             return ret
 
     if build:
+        # get the functions default value and args
+        argspec = salt.utils.args.get_function_argspec(__salt__['docker.build'])
+        # Map any if existing args from kwargs into the build_args dictionary
+        build_args = dict(zip(argspec.args, argspec.defaults))
```
Comment (rallytime) [inline salt/states/docker_image.py]: And then change this to:
```
build_args = dict(list(zip(argspec.args, argspec.defaults)))
```

**Review by rallytime — CHANGES_REQUESTED** [review]
There are a couple of pylint errors here. I've commented below with the needed changes. :)

**Comment by rallytime** [comment]
Hi @pabloh007 and welcome! The pylint failures reported are for Python 3 compatibility issues, and those are a little harder to figure out to spot the error with our lint report. Once you get those fixed up we can review this again. Thanks!

**Comment by pabloh007** [comment]
hey @rallytime  Ill work on those changes.

**Comment by rallytime** [comment]
re-run lint

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
@terminalmage re-review requested

**Comment by cachedout** [comment]
@pabloh007 I apologize that this has taken a little while for us to get back to. This has some merge conflicts. Could you please resolve them and push up a fix? Thanks.

@terminalmage We need a very quick re-review from you. Should only take a moment since your concerns have been resolved. Thanks.

**Comment by pabloh007** [comment]
Have resolved conflicts @cachedout . Thanks!

**Review by terminalmage — APPROVED** [review]

**Comment by rallytime** [comment]
Thanks @pabloh007!

**Pull request merged**